include/ozo/connection_info.h

```diff
@@ -30,7 +30,7 @@ class connection_info {
     Statistics statistics;
 
 public:
-    using connection_type = std::shared_ptr<ozo::connection<OidMap, Statistics>>; //!< Type of connection which is produced by the source.
+    using connection_type = std::shared_ptr<ozo::connection<std::decay_t<OidMap>, Statistics>>; //!< Type of connection which is produced by the source.
```

Comment:
Should we check this by tests? Use static_assert or just apply same aproach as in the snippet somewhere.